pyam/core.py

```diff
@@ -757,26 +758,30 @@ def rename(self, mapping=None, inplace=False, append=False,
         if not inplace:
             return ret
 
-    def convert_unit(self, conversion_mapping, inplace=False):
-        """Converts units based on provided unit conversion factors
+    def convert_unit(self, current, to=None, factor=None, inplace=False):
+        """Converts a unit using given factor or pint.UnitRegistry
```

Comment:
It might be good to put a line about when we can expect pint to produce the correct answers

Reply:
not quite sure what you mean here - would a link to the pint docs page be sufficient?

Reply:
As in, some description of when you will need to hard code a conversion factor when the conversion is context-dependent (Mt CO2-equiv in various frames) and when you won't (kt -> Gt etc).

Reply:
added two paragraph using `EJ` & `GWh` as examples, also mentioning combined units `EJ/yr`, plus a link to the pint docs and the energy-units repo.

`Mt CO2-equiv` is a too specific use case for the general docs, in my view... maybe something for another PR with some context- and custom-UnitRegistry-handling...